src/drivers/intel/cavs/ssp.c

```diff
@@ -871,6 +872,15 @@ static int ssp_probe(struct dai *dai)
 {
 	struct ssp_pdata *ssp;
 
+	if (dai_get_drvdata(dai))
+		return 0; /* already probed */
+#if defined(CONFIG_APOLLOLAKE)
+	/* Disable dynamic clock gating for i2s before touching any register */
```

Comment:
Again this would use the pm_runtime get/put which would compile out on baytrail or platforms where its not defined.